### DISCUSSION TITLE
Stream sometimes in "idle" for no reason. What should I do?

### DISCUSSION THREAD

**Discussion by al3x88** [HaishinKit/HaishinKit.swift#1646] [Q&A]
Hi,
I noticed that while streaming something the HKStreamReadyState goes in ".idle", so I guess it stops the streaming.
Do you guys knows what cause the stream to go in idle state?

In my code I implemented a monitor every 15sec as follows:

```
guard let stream = await self.netStreamSwitcher.stream else {
                return
            }

            
            let status = await stream.readyState
            
            if status == .idle {
                self.resetRTMPConn()
            }
```

Where self.resetRTMPConn() is:

    func resetRTMPConn(){
        Task{ 
                        
            await self.netStreamSwitcher.close()
            
            await self.netStreamSwitcher.open(.ingest)
           
        }
    }

This trick saved me since at some point it starts to stream again, but other times takes too many attempts.
Do you guys have a better approach?
What are the root cause of HKStreamReadyState in idle state?

thanks a lot!!!

**Comment by shogo4405** [comment]
The publish state may transition to idle in the following cases:
1. When there is no response from the server within a certain period at the start of streaming.
1. When the server disconnects for some reason during streaming.

To reproduce this issue locally, logs with the trace level can be helpful:
```swift
LBLogger.with(kHaishinKitIdentifier).level = .trace
```

Handling can be implemented using the status property in RTMPConnection and RTMPStream:
- https://github.com/shogo4405/HaishinKit.swift/blob/9308ed18e6081e295e752522ed979a7ecb3e9f62/HaishinKit/Sources/RTMP/RTMPConnection.swift#L155-L160
- https://github.com/shogo4405/HaishinKit.swift/blob/9308ed18e6081e295e752522ed979a7ecb3e9f62/HaishinKit/Sources/RTMP/RTMPStream.swift#L185-L190

**Reply by al3x88** [reply]
@shogo4405 is it possibile to save those logs locally in the device? 
Would be useful to get them from other devices in case of issues....

**Comment by al3x88** [comment]
@shogo4405 forcing connection reset like this:

`
  await self.netStreamSwitcher.close()
      
await self.netStreamSwitcher.open(.ingest)
`

is something that makes sense?
Or the connection already attempts to re-establish the connection, in case of idle?

**Reply by shogo4405** [reply]
I'm stopped providing technical support. Please look it up on your own. Using the readyState approach is a bad idea, so you should avoid it.
